name: CI Test

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

permissions:
  contents: read
  issues: write
  
jobs:
  ci-build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3

      - name: Create issue for direct push
        if: github.ref == 'refs/heads/main' && github.event.head_commit.committer.name != 'GitHub'
        uses: dacbd/create-issue-action@main
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          title: "Direct push to main: ${{ github.actor }}"
          body: |
            Whoops! Looks like **${{ github.actor }}** pushed directly to the main branch! ðŸ’€
            I'm sure this was just a innocent mistake ðŸ™ƒ
            - Message: ${{ github.event.head_commit.message }}
            - URL: ${{ github.event.head_commit.url }}
          labels: whoops